inmemory/src/main/java/org/vertexium/inmemory/InMemoryGraph.java

```diff
@@ -308,6 +308,11 @@ public Edge save(Authorizations authorizations) {
     private Edge savePreparedEdge(final EdgeBuilderBase edgeBuilder, final String outVertexId, final String inVertexId, Long timestamp, Authorizations authorizations) {
         if (timestamp == null) {
             timestamp = IncreasingTime.currentTimeMillis();
+
+            // The timestamps will be incremented below, this will ensure future mutations will be in the future
+            for (int i = 0; i < 10; i++) {
+                IncreasingTime.currentTimeMillis();
+            }
```

Comment:
Maybe we should create a new method in `IncreasingTime` named something like `advanceLastTime`. The code itself would be a little more explicit about what's happening. This loop plus the bonus call in core/src/main/java/org/vertexium/mutation/AlterPropertyVisibility.java could be replaced with a call to the new method.